sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -2050,4 +2051,93 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       )
     )
   }
+
+  // scalastyle:off nonascii
+  test("UTF8String validation") {
+    def isValidUTF8(str: Any, expected: Any): Unit = {
+      checkEvaluation(IsValidUTF8(Literal.create(str, StringType)), expected)
+    }
+    isValidUTF8(null, null)
+    isValidUTF8("", true)
+    isValidUTF8("aa", true)
+    isValidUTF8("\u0061", true)
+    isValidUTF8(UTF8String.EMPTY_UTF8, true)
+    isValidUTF8(UTF8String.fromString(""), true)
+    isValidUTF8(UTF8String.fromString("abc"), true)
+    isValidUTF8(UTF8String.fromString("hello"), true)
+    isValidUTF8(UTF8String.fromBytes(Array.empty[Byte]), true)
+    isValidUTF8(UTF8String.fromBytes(Array[Byte](0x41)), true)
+    isValidUTF8(UTF8String.fromBytes(Array[Byte](0x61)), true)
+    isValidUTF8(UTF8String.fromBytes(Array[Byte](0x80.toByte)), false)
+    isValidUTF8(UTF8String.fromBytes(Array[Byte](0xFF.toByte)), false)
+
+    def makeValidUTF8(str: Any, expected: String): Unit = {
+      checkEvaluation(
+        MakeValidUTF8(Literal.create(str, StringType)),
+        UTF8String.fromString(expected)
+      )
+    }
+    makeValidUTF8(null, null)
+    makeValidUTF8("", "")
+    makeValidUTF8("aa", "aa")
+    makeValidUTF8("\u0061", "\u0061")
+    makeValidUTF8(UTF8String.EMPTY_UTF8, "")
+    makeValidUTF8(UTF8String.fromString(""), "")
+    makeValidUTF8(UTF8String.fromString("abc"), "abc")
+    makeValidUTF8(UTF8String.fromString("hello"), "hello")
+    makeValidUTF8(UTF8String.fromBytes(Array.empty[Byte]), "")
+    makeValidUTF8(UTF8String.fromBytes(Array[Byte](0x41)), "A")
+    makeValidUTF8(UTF8String.fromBytes(Array[Byte](0x61)), "a")
+    makeValidUTF8(UTF8String.fromBytes(Array[Byte](0x80.toByte)), "\ufffd")
```

Comment:
```suggestion
    makeValidUTF8(UTF8String.fromBytes(Array[Byte](0x80.toByte)), "\uFFFD")
```
nit: I find it easier to read a UNICODE character if the hex letters are capital (different case from the leading `u`).